source/acquaman/AMGenericContinuousScanConfiguration.cpp

```diff
@@ -0,0 +1,328 @@
+	}
+
+	computeTotalTime();
+}
+
+void AMGenericContinuousScanConfiguration::removeControl(int axisId)
+{
+	if (axisId < axisControlInfos_.count() && axisId >= 0){
+
+		AMScanAxis *axis = removeScanAxis(axisId);
+		axis->regionAt(0)->disconnect();
+		axis->regionAt(0)->deleteLater();
+		axis->deleteLater();
+		axisControlInfos_.remove(axisId);
+		setModified(true);
+	}
+}
+
+void AMGenericContinuousScanConfiguration::addDetector(AMDetectorInfo newInfo)
+{
+	bool containsDetector = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size; i++){
+
+		if (newInfo.name() == detectorConfigurations_.at(i).name())
+			containsDetector = true;
+	}
+
+	if (!containsDetector){
+
+		detectorConfigurations_.append(newInfo, newInfo.name());
+		setModified(true);
+	}
+}
+
+void AMGenericContinuousScanConfiguration::removeDetector(AMDetectorInfo info)
+{
+	bool detectorRemoved = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size && !detectorRemoved; i++){
+
+		if (info.name() == detectorConfigurations_.at(i).name()){
+
+			detectorRemoved = true;
+			detectorConfigurations_.remove(i);
+			setModified(true);
+		}
+	}
+}
+
+void AMGenericContinuousScanConfiguration::addRegionOfInterest(AMRegionOfInterest *region)
+{
+	regionsOfInterest_.append(region);
+	setModified(true);
+}
+
+void AMGenericContinuousScanConfiguration::removeRegionOfInterest(AMRegionOfInterest *region)
+{
+	foreach (AMRegionOfInterest *regionToBeRemoved, regionsOfInterest_)
+		if (regionToBeRemoved->name() == region->name()){
+
+			regionsOfInterest_.removeOne(regionToBeRemoved);
+			setModified(true);
+		}
+}
+
+void AMGenericContinuousScanConfiguration::setI0(const AMDetectorInfo &info)
+{
+	bool nameInDetectorList = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size && !nameInDetectorList; i++)
+		if (info.name() == detectorConfigurations_.at(i).name())
+			nameInDetectorList = true;
+
+	if (nameInDetectorList){
+
+		i0_ = info;
+		setModified(true);
+	}
+}
+
+AMDbObjectList AMGenericContinuousScanConfiguration::dbReadRegionsOfInterest()
+{
+	AMDbObjectList listToBeSaved;
+
+	foreach (AMRegionOfInterest *region, regionsOfInterest_)
+		listToBeSaved << region;
+
+	return listToBeSaved;
+}
+
+void AMGenericContinuousScanConfiguration::dbLoadRegionsOfInterest(const AMDbObjectList &newRegions)
+{
+	regionsOfInterest_.clear();
+
+	foreach (AMDbObject *newObject, newRegions){
+
+		AMRegionOfInterest *region = qobject_cast<AMRegionOfInterest *>(newObject);
+
+		if (region)
+			regionsOfInterest_.append(region);
+	}
+}
+
+QString AMGenericContinuousScanConfiguration::regionsOfInterestHeaderString(const QList<AMRegionOfInterest *> &regions) const
+{
+	QString string = "";
+
+	if (!regions.isEmpty()){
+
+		string.append("\nRegions Of Interest\n");
+
+		foreach (AMRegionOfInterest *region, regions)
+			string.append(QString("%1\t%2 eV\t%3 eV\n").arg(region->name()).arg(region->lowerBound()).arg(region->upperBound()));
```

Comment:
Did a quick search. We are composing the same string in several function. Maybe we should has a function (like `toHeaderString()` or something similar) in `AMRegionOfInterest` to compose the string.